smallrye-reactive-messaging-provider/src/main/java/io/smallrye/reactive/messaging/PublisherMediator.java

```diff
@@ -110,17 +110,33 @@ private void produceAPublisherOfMessages() {
     }
 
     private void produceIndividualMessages() {
-        setPublisher(ReactiveStreams.generate(() -> {
-            Message message = invoke();
-            Objects.requireNonNull(message,
-                    "The method " + configuration.methodAsString() + " returned an invalid value: null");
-            return message;
-        }));
+        if (configuration.isBlocking()) {
+            setPublisher(ReactiveStreams.<Uni<?>> generate(this::invokeBlocking)
+                    .flatMapCompletionStage(Uni::subscribeAsCompletionStage)
+                    .map(message -> {
+                        Objects.requireNonNull(message,
```

Comment:
This should have been tested before (I've added a comment about this case).

Reply:
So remove the null check?

Reply:
If you follow the other approach in `invokeBlocking` yes.